docs/DeveloperGuide.md

```diff
@@ -4,10 +4,137 @@
+it into a `CommandType` enum.<br>
+The `splitInput` method is used to split the user input into an array of strings, according to the command type.
+It returns an array containing any arguments that are needed to execute the command.
+
+### Logic Component
+The logic component consists of classes that handle the logic of the application. The logic component is divided into
+`MainLogic` and `SubLogic` which consists of `OrderLogic` and `MenuLogic`.
+
+* [**`MainLogic`**](#model-component): A class to handle the first level commands, and pass the user input to corresponding
+classes for analysis and execution.
+* [**`OrderLogic`**](#model-component) and [**`MenuLogic`**](#model-component): A class to handle the second level commands,
+and pass the user input to corresponding classes for analysis and execution.
+
+### Command Component
+The command component consists of two different command interfaces: `MainCommand` and `OrderCommand`. The `MainCommand`
+interface is for the various command classes that are used in the `MainLogic`, while the `OrderCommand` interface is for
+command classes used in `OrderLogic`.  
+
+A `Command` object will be created in either the `MainLogic` or `OrderLogic`
+based on what command the user has inputted(e.g., `MainHelpCommand` object is created when the user inputs the `help`
+command). The `execute()` method of the `Command` object is then called to execute the command, which may require
+certain arguments based on the type of command.
+
+### Model Component
+The model consists of classes describing the objects used in this application.
+The general structure is that menu and order are separate, but they both work with `menuItem(s)`, which 
+represent food items on the menu.
+
+* [**`ItemManager`**](#model-component): An interface containing methods representing operations common to **`Menu`** 
+  and **`Order`**. <br><br />
+* [**`Item`**](#model-component): An abstract class representing a food item. It should be implemented by **`MenuItem`**.
+  <br><br />
+* [**`Menu`**](#model-component): A class representing the menu(s) of the restaurant, where each contains menuItem(s)
+ that can be ordered. Multiple menus can exist and each has a unique ID. <br><br />
+* [**`MenuItem`**](#model-component): A class inheriting item, and represents a food item on the menu. <br><br />
+* [**`Order`**](#model-component): A class representing an order to be entered into the system to be kept track of. Each 
+  order has a unique ID generated from the time of order.<br><br />
+* [**`SetMenu`**](#model-component): An enumeration representing the different types of set menus available, examples of
+  which includes *breakfast*, *lunch*, *dinner*.
+
+The *Class Diagram* below shows how the model components interact with each other, including interactions such as 
+dependencies, associations and inheritance.
+
+![Class Diagram](images/modelcomponent.png)
```

Comment:
<img width="212" alt="image" src="https://github.com/nus-cs2113-AY2324S2/tp/assets/48584322/5f43432b-ae8b-4f50-8c8b-60728e18fd4a">

Will it be better to remove the icon as it is not taught in class? It can be disabled 😄